kratos/geometries/nurbs_curve_on_surface_geometry.h

```diff
@@ -0,0 +1,294 @@
+        : BaseType(rOther, &msGeometryData)
+        , mpNurbsSurface(rOther.mpNurbsSurface)
+        , mpNurbsCurve(rOther.mpNurbsCurve)
+    {
+    }
+
+    /// Destructor
+    ~NurbsCurveOnSurfaceGeometry() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * Assignment operator.
+     *
+     * @note This operator don't copy the points and this
+     * geometry shares points with given source geometry. It's
+     * obvious that any change to this geometry's point affect
+     * source geometry's points too.
+     *
+     * @see Clone
+     * @see ClonePoints
+     */
+    NurbsCurveOnSurfaceGeometry& operator=(const NurbsCurveOnSurfaceGeometry& rOther)
+    {
+        BaseType::operator=(rOther);
+        mpNurbsSurface = rOther.mpNurbsSurface;
+        mpNurbsCurve = rOther.mpNurbsCurve;
+        return *this;
+    }
+
+    /**
+     * @brief Assignment operator for geometries with different point type.
+     *
+     * @note This operator don't copy the points and this
+     * geometry shares points with given source geometry. It's
+     * obvious that any change to this geometry's point affect
+     * source geometry's points too.
+     *
+     * @see Clone
+     * @see ClonePoints
+     */
+    template<class TOtherCurveContainerPointType, class TOtherSurfaceContainerPointType>
+    NurbsCurveOnSurfaceGeometry& operator=(
+        NurbsCurveOnSurfaceGeometry<TWorkingSpaceDimension, TOtherCurveContainerPointType, TOtherSurfaceContainerPointType> const & rOther)
+    {
+        BaseType::operator=(rOther);
+        mpNurbsSurface = rOther.mpNurbsSurface;
+        mpNurbsCurve = rOther.mpNurbsCurve;
+        return *this;
+    }
+
+    ///@}
+    ///@name OperationsZeroVector(3);
+    ///@{
+
+    /*typename BaseType::Pointer Create(
+        PointsArrayType const& ThisPoints) const override
+    {
+        return Kratos::make_shared<NurbsCurveOnSurfaceGeometry>(ThisPoints);
+    }*/
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /*
+    * @brief This method maps from dimension space to working space.
+    * From Piegl and Tiller, The NURBS Book, Algorithm A3.1/ A4.1
+    * @param rResult array_1d<double, 3> with the coordinates in working space
+    * @param LocalCoordinates The local coordinates in dimension space
+    * @return array_1d<double, 3> with the coordinates in working space
+    * @see PointLocalCoordinates
+    */
+    CoordinatesArrayType& GlobalCoordinates(
+        CoordinatesArrayType& rResult,
+        const CoordinatesArrayType& rLocalCoordinates
+    ) const override
+    {
+        // Compute the coordinates of the embedded curve in the parametric space of the surface
+        rResult = mpNurbsCurve->GlobalCoordinates(rResult, rLocalCoordinates);
+        
+        // Compute and return the coordinates of the surface in the geometric space
+        return mpNurbsSurface->GlobalCoordinates(rResult, rResult);
```

Comment:
Be careful that now your input and output are the same. So the use of noalias in the GlobalCoordiantes method would give wrong results.

Reply:
+1 @pooyan-dadvand thanks for the observation!!